petisco/extra/sqlalchemy/sql/base_sql_repository.py

```diff
@@ -0,0 +1,52 @@
+from meiga import BoolResult, Error, Failure, isSuccess
+
+from petisco import Uuid
+from petisco.base.application.patterns.repository import Repository
+from petisco.base.domain.errors.defaults.already_exists import (
+    AggregateAlreadyExistError,
+)
+from petisco.base.domain.errors.defaults.not_found import (
+    AggregateNotFoundError,
+    AggregatesNotFoundError,
+)
+
+
+class BaseSqlRepository(Repository):
+    @classmethod
+    def fail_if_entity_already_exist(
```

Comment:
maybe we should use aggregate instead of entity here as well. This will be a breaking change, but it worth it.

Reply:
Yes, this is more consistent